neqo-transport/src/connection/mod.rs

```diff
@@ -2204,94 +2200,6 @@ impl Connection {
         Ok(())
     }
 
-    fn validate_cids(&mut self) -> Res<()> {
-        match self.version() {
-            QuicVersion::Draft27 => self.validate_cids_draft_27(),
-            _ => self.validate_cids_draft_28_plus(),
-        }
-    }
-
-    fn validate_cids_draft_27(&mut self) -> Res<()> {
-        if let AddressValidationInfo::Retry { token, .. } = &self.address_validation {
-            debug_assert!(!token.is_empty());
-            let tph = self.tps.borrow();
-            let tp = tph
-                .remote
-                .as_ref()
-                .unwrap()
-                .get_bytes(tparams::ORIGINAL_DESTINATION_CONNECTION_ID);
-            if self
-                .original_destination_cid
-                .as_ref()
-                .map(ConnectionId::as_cid_ref)
-                != tp.map(ConnectionIdRef::from)
-            {
-                return Err(Error::InvalidRetry);
-            }
-        }
-        Ok(())
-    }
-
-    fn validate_cids_draft_28_plus(&mut self) -> Res<()> {
```

Comment:
you should not remove this validation. This is still needed for draft 29 to 32 and for final version 1.
You should only remove validate_cids_draft_27.
This is a reason why the test mitm_retry is failing

Reply:
Yep, this is it.  Rename this function as validate_cids(), I think.

Reply:
Put the function back, renamed it validate_cids() and added the call to the function back to the top of the process_tps(). All tests are now passing on my machine. Squashed my commits, believe its ready!